src/ui/SwiftBuildStatus.ts

```diff
@@ -0,0 +1,158 @@
+            if (!configuration.showBuildStatus) {
+                return;
+            }
+            this.handleTaskStatus(event.execution.task);
+        });
+    }
+
+    dispose() {
+        this.onDidStartTaskDisposible.dispose();
+    }
+
+    private handleTaskStatus(task: vscode.Task): void {
+        // Only care about swift tasks
+        if (!(task.execution && task.execution instanceof SwiftExecution)) {
+            return;
+        }
+        // Default to setting if task doesn't overwrite
+        const showBuildStatus: ShowBuildStatusOptions =
+            task.definition.showBuildStatus || configuration.showBuildStatus;
+        if (showBuildStatus === "never") {
+            return;
+        }
+
+        const execution = task.execution as SwiftExecution;
+        const disposables: vscode.Disposable[] = [];
+        const handleTaskOutput = (update: (message: string) => void) =>
+            new Promise<void>(res => {
+                const done = () => {
+                    disposables.forEach(d => d.dispose());
+                    res();
+                };
+                disposables.push(
+                    execution.onDidWrite(data => {
+                        if (this.parseEvents(task, data, update)) {
+                            done();
+                        }
+                    }),
+                    execution.onDidClose(done)
+                );
+            });
+        if (showBuildStatus === "progress" || showBuildStatus === "notification") {
+            vscode.window.withProgress<void>(
+                {
+                    location:
+                        showBuildStatus === "progress"
+                            ? vscode.ProgressLocation.Window
+                            : vscode.ProgressLocation.Notification,
+                },
+                progress => handleTaskOutput(message => progress.report({ message }))
+            );
+        } else {
+            this.statusItem.showStatusWhileRunning(task, () =>
+                handleTaskOutput(message => this.statusItem.update(task, message))
+            );
+        }
+    }
+
+    /**
+     * @param data
+     * @returns true if done, false otherwise
+     */
+    private parseEvents(
+        task: vscode.Task,
+        data: string,
+        update: (message: string) => void
+    ): boolean {
+        const sanitizedData = stripAnsi(data);
+        // We'll process data one line at a time, in reverse order
+        // since the latest interesting message is all we need to
+        // be concerned with
+        const lines = sanitizedData.split(/\r\n|\n|\r/gm).reverse();
+        for (const line of lines) {
+            if (this.checkIfBuildComplete(line)) {
+                return true;
+            }
+            const progress = this.findBuildProgress(line);
+            if (progress) {
+                update(`Building "${task.name}" (${progress.completed}/${progress.total})`);
```

Comment:
It looks much cleaner without the `Building` text. `Building "Build All"` just sounds weird. Also would be nice to have the text be consistent with the original text from the status item which is constructed as follows
```
const folder = task.definition.cwd as string;
if (folder) {
    return `${task.name} (${path.basename(folder)})`;
} else {
    return task.name;
}
```